nailgun/entities.py

```diff
@@ -847,6 +847,57 @@ def update(self, fields=None):
         return self.read()
 
 
+class VMWareComputeResource(AbstractComputeResource):
+    # pylint: disable=too-many-ancestors
+    """A representation for compute resources with Vmware provider
+    """
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'set_console_password': entity_fields.BooleanField(),
+            'user': entity_fields.StringField(),
+            'password': entity_fields.StringField(),
+            'datacenter': entity_fields.StringField()
+        }
+        super(VMWareComputeResource, self).__init__(server_config, **kwargs)
+        self._fields['provider'].default = 'Vmware'
+        self._fields['provider'].required = True
+        self._fields['provider_friendly_name'].default = 'VMware'
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        if ignore is None:
+            ignore = set()
+
+        ignore.add('password')
+
```

Comment:
nitpick: empty lines aren't adding readability here since the method is 5 lines long, please remove them